support-frontend/assets/pages/contributions-landing/components/StripeCardForm/StripeCardForm.jsx

```diff
@@ -337,15 +378,19 @@ const CardForm = (props: PropTypes) => {
   }, [props.setupIntentClientSecret]);
 
   useEffect(() => {
-    const formIsComplete =
+    let formIsComplete =
       fieldStates.CardNumber.name === 'Complete' &&
       fieldStates.Expiry.name === 'Complete' &&
       fieldStates.CVC.name === 'Complete';
 
+    if (props.zipCodeFieldStatus === 'REQUIRED') {
+      formIsComplete = formIsComplete && isValidZipCode(zipCode);
+    }
```

Comment:
question: If the status is `OPTIONAL`, should we still run validation if they've entered something into the field?

Reply:
hmm, we shouldn't give it to stripe if it's wrong, but I'm not sure if it should block submission

Reply:
I just had a look at the default behaviour of `react-hook-form` (might be worth adding to this project at some point?). If you mark a field as optional, but provide incorrect data, it will validate it and prevent submission. Perhaps that is the behaviour we should go for?

Reply:
100% agree we shouldn't send bad data to stripe regardless (currently the code here *will* send bad data to stripe)

Reply:
Yes that sounds right to me